File: app/src/main/java/ai/kitt/vnest/util/KeyboardUtil.java
package ai.kitt.vnest.util;

import android.annotation.SuppressLint;
import android.app.Activity;
import android.view.View;
import android.view.ViewGroup;
import android.view.inputmethod.InputMethodManager;
import android.widget.EditText;

import androidx.appcompat.app.AppCompatDialogFragment;
import androidx.fragment.app.FragmentActivity;

public class KeyboardUtil {

    public static void hideSoftKeyboard(Activity activity, View view) {
        if (activity == null) return;
        InputMethodManager inputMethodManager =
                (InputMethodManager) activity.getSystemService(
                        Activity.INPUT_METHOD_SERVICE);
        if(view!=null) {
            inputMethodManager.hideSoftInputFromWindow(view.getWindowToken(), 0);
        }

    }

    public static void hideSoftKeyboard(Activity activity) {
        hideSoftKeyboard(activity,activity.getCurrentFocus());
    }

    public static void hideSoftKeyboard(AppCompatDialogFragment dialogFragment) {
        if (dialogFragment == null) return;
        FragmentActivity activity = dialogFragment.getActivity();
        if (activity == null) return;
        InputMethodManager inputMethodManager = (InputMethodManager) activity.getSystemService(Activity.INPUT_METHOD_SERVICE);
        View currentFocus = dialogFragment.getView();
        if (currentFocus != null) {
            inputMethodManager.hideSoftInputFromWindow(currentFocus.getWindowToken(), 0);
        }
    }

    @SuppressLint("ClickableViewAccessibility")
    public static void setupHideKeyboarOutTouchScreen(Activity activity, View view) {
        if (activity == null) return;
        if (view == null) return;
        // Set up touch listener for non-text box views to hide keyboard.
        if (!(view instanceof EditText)) {
            view.setOnTouchListener((v, event) -> {
                KeyboardUtil.hideSoftKeyboard(activity);
                return false;
            });
        }

        //If a layout container, iterate over children and seed recursion.
        if (view instanceof ViewGroup) {
            for (int i = 0; i < ((ViewGroup) view).getChildCount(); i++) {
                View innerView = ((ViewGroup) view).getChildAt(i);
                setupHideKeyboarOutTouchScreen(activity, innerView);
            }
        }
    }

    @SuppressLint("ClickableViewAccessibility")
    public static void setupHideKeyboarOutTouchScreen(AppCompatDialogFragment dialogFragment, View view) {
        if (dialogFragment == null) return;
        if (view == null) return;
        // Set up touch listener for non-text box views to hide keyboard.
        if (!(view instanceof EditText)) {
            view.setOnTouchListener((v, event) -> {
                KeyboardUtil.hideSoftKeyboard(dialogFragment);
                return false;
            });
        }

        //If a layout container, iterate over children and seed recursion.
        if (view instanceof ViewGroup) {
            for (int i = 0; i < ((ViewGroup) view).getChildCount(); i++) {
                View innerView = ((ViewGroup) view).getChildAt(i);
                setupHideKeyboarOutTouchScreen(dialogFragment, innerView);
            }
        }
    }
}

File: baseui/src/main/java/roxwin/tun/baseui/utils/KeyboardUtil.kt
package roxwin.tun.baseui.utils

import android.app.Activity
import android.content.Context
import android.view.MotionEvent
import android.view.View
import android.view.View.OnTouchListener
import android.view.ViewGroup
import android.view.inputmethod.InputMethodManager
import android.widget.EditText
import androidx.fragment.app.DialogFragment

object KeyboardUtil {
    @JvmStatic
    fun hideSoftKeyboard(activity: Activity?) {
        if (activity == null) return
        val inputMethodManager =
            activity.getSystemService(
                Activity.INPUT_METHOD_SERVICE
            ) as InputMethodManager
        val currentFocus = activity.currentFocus
        if (currentFocus != null) {
            inputMethodManager.hideSoftInputFromWindow(currentFocus.windowToken, 0)
        }
    }
    @JvmStatic
    fun hideSoftKeyboard(dialogFragment: DialogFragment?) {
        if (dialogFragment == null) return
        val activity = dialogFragment.activity ?: return
        val inputMethodManager =
            activity.getSystemService(
                Activity.INPUT_METHOD_SERVICE
            ) as InputMethodManager
        val currentFocus = dialogFragment.view
        if (currentFocus != null) {
            inputMethodManager.hideSoftInputFromWindow(currentFocus.windowToken, 0)
        }
    }

    @JvmStatic
    fun hideKeyboardFrom(context: Context, view: View) {
        val imm = context.getSystemService(Activity.INPUT_METHOD_SERVICE) as InputMethodManager
        imm.hideSoftInputFromWindow(view.windowToken, 0)
    }
    fun setupHideKeyboarOutTouchScreen(activity: Activity?, view: View?) {
        if (activity == null) return
        if (view == null) return
        // Set up touch listener for non-text box views to hide keyboard.
        if (view !is EditText) {
            view.setOnTouchListener(OnTouchListener { v: View?, event: MotionEvent? ->
                hideSoftKeyboard(activity)
                false
            })
        }

        //If a layout container, iterate over children and seed recursion.
        if (view is ViewGroup) {
            for (i in 0 until view.childCount) {
                val innerView = view.getChildAt(i)
                setupHideKeyboarOutTouchScreen(
                    activity,
                    innerView
                )
            }
        }
    }

    fun setupHideKeyboarOutTouchScreen(
        dialogFragment: DialogFragment?,
        view: View?
    ) {
        if (dialogFragment == null) return
        if (view == null) return
        // Set up touch listener for non-text box views to hide keyboard.
        if (view !is EditText) {
            view.setOnTouchListener(OnTouchListener { v: View?, event: MotionEvent? ->
                hideSoftKeyboard(
                    dialogFragment
                )
                false
            })
        }

        //If a layout container, iterate over children and seed recursion.
        if (view is ViewGroup) {
            for (i in 0 until view.childCount) {
                val innerView = view.getChildAt(i)
                setupHideKeyboarOutTouchScreen(
                    dialogFragment,
                    innerView
                )
            }
        }
    }
}